core/arch/arm/mm/tee_pager.c

```diff
@@ -176,12 +177,14 @@ static struct tee_pager_area *tee_pager_find_area(vaddr_t va)
 	return NULL;
 }
 
-static uint32_t get_area_mattr(struct tee_pager_area *area __unused)
+static uint32_t get_area_mattr(struct tee_pager_area *area)
 {
 	uint32_t attr = TEE_MATTR_VALID_BLOCK | TEE_MATTR_GLOBAL |
 			TEE_MATTR_CACHE_DEFAULT | TEE_MATTR_SECURE;
 
-	attr |= TEE_MATTR_PRWX;
+	attr |= TEE_MATTR_PRW;
+	if (area->flags & TEE_PAGER_AREA_X)
```

Comment:
Why not support readonly pages also? It certainly make sense for paged pages, or was there a problem?